### PR TITLE
Introduce markdown lint check

### LINKED ISSUE #7129: Introduce markdown-lint check
Just like what we did in https://github.com/open-telemetry/opentelemetry-specification/blob/b1d14b599970ee4771e572bafdd8c3d8a4a4aaf3/.github/workflows/checks.yaml#L19

### PR DESCRIPTION
**Pull request by aaron-ai**
Fixes #7129

### CODE REVIEW

File: instrumentation/spring/README.md [hunk 1]
```diff
@@ -841,14 +837,14 @@ To configure OpenTelemetry tracing with the OTLP, Zipkin, or Jaeger span exporte
     <groupId>io.opentelemetry.instrumentation</groupId>
     <artifactId>opentelemetry-zipkin-spring-boot-starter</artifactId>
     <version>OPENTELEMETRY_VERSION</version>
-</dependency>
+</dependency>
```
Comment (trask) [inline instrumentation/spring/README.md:840]: do you know why the trailing space is relevant?

File: .markdownlint.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+default: true
+
+ul-style: false
+line-length: false
+no-duplicate-header:
+  siblings_only: true
+ol-prefix:
+  style: ordered
+no-inline-html: false
+fenced-code-language: false
```
Comment (trask) [inline .markdownlint.yml:14]: matches spec repo 👍

File: instrumentation/jdbc/library/README.md [hunk 3]
```diff
@@ -65,14 +65,14 @@ public class DataSourceConfig {
 
 #### Driver way
 
-1. Activate tracing for JDBC connections by setting `jdbc:otel:` prefix to the JDBC URL:
+* Activate tracing for JDBC connections by setting `jdbc:otel:` prefix to the JDBC URL:
```
Comment (mateuszrzeszutek) [inline instrumentation/jdbc/library/README.md:68]: Can we go back to the ordered list here? I think these points are meant to be instructions that you should execute in order.

**Review by mateuszrzeszutek — APPROVED** [review]
Looks great, thanks!

File: instrumentation/jdbc/library/README.md (hunk 3 above)
Comment (aaron-ai) [reply instrumentation/jdbc/library/README.md]: OK, I fixed it to pass the markdown-lint check

**Review by mateuszrzeszutek — APPROVED** [review]

File: instrumentation/spring/README.md (hunk 1 above)
Comment (trask) [reply instrumentation/spring/README.md:840]: it looks like the spotless check is failing on the trailing space

Comment (aaron-ai) [reply instrumentation/spring/README.md]: fixed

**Pull request merged**